wagtail/admin/tests/test_admin_search.py

```diff
@@ -107,3 +109,78 @@ def test_search_other(self):
 
         self.assertNotIn("Pages", rendered)
         self.assertIn("My Search", rendered)
+
+
+class TestCase(BaseSearchAreaTestCase):
+    """Tests the comparison functions."""
+
+    def setUp(self):
+        self.search_area1 = SearchArea("Label 1", "/url1", order=100)
+        self.search_area2 = SearchArea("Label 2", "/url2", order=200)
+        self.search_area3 = SearchArea("Label 1", "/url3", order=300)
+
+    def test_eq(self):
+        self.assertEqual(
+            self.search_area1, self.search_area1
+        )  # Same object, should be equal
+        self.assertEqual(
+            self.search_area1, self.search_area1
+        )  # Same label and order, should be equal
+        self.assertNotEqual(
+            self.search_area1, self.search_area2
+        )  # Different order, should not be equal
```

Comment:
We also need to test the inequality of a `SearchArea` object with a non-`SearchArea` object, e.g. a string

Reply:
Not quite sure how to get this done, but giving it a try.

```
def test_eq(self):
        # Same label and order, should be equal
        self.assertTrue(self.search_area1 == self.search_area1)

        # Different order, should not be equal
        self.assertFalse(self.search_area1 == self.search_area2)

        # Inequality with a non-SearchArea object, such as a string
        non_searcharea = "Not a SearchArea object"
        self.assertNotEqual(
            self.search_area1,
            non_searcharea,
            "Expected inequality with non-SearchArea object",
        )
```
OR
```
def test_inequality_with_non_searcharea(self):
    # Inequality with a non-SearchArea object, such as a string
    non_searcharea = "Not a SearchArea object"
    self.assertNotIsInstance(
        self.search_area1, type(non_searcharea), "Expected inequality with non-SearchArea object"
    )
```

In the first snippet the inequality is being checked right in the `test_eq` function, while in the second snippet there's a second function `test_inequality_with_non_searcharea` for the same.

Please let me know if this can work, or else some more information or example can be helpful.